File: darts-java/src/main/java/com/stride/scorables/Single.java
package com.stride.scorables;

public class Single implements Scorable {
    private int score;

    public Single(int score) {
        if (score < 1 || score > 20) {
            throw new IllegalArgumentException("Throw score must be in range 1..20");
        }
        this.score = score;
    }

    public int score() {
        return score;
    }
}

File: javascript/app/darts/bed/single.js
class Single {
    constructor(bedValue) {
        if (bedValue < 1) {
            throw new RangeError('Bed score must be in range 1..20');
        }
        this.bedValue = bedValue;
    }

    bedScore() {
        return this.bedValue;
    }
}

module.exports = Single;